agent/recorder.go

```diff
@@ -167,19 +188,53 @@ func (r *SpanRecorder) SendSpans() (error, bool) {
 	}
 
 	if payloadSent {
-		r.okSend++
+		atomic.AddInt64(&r.stats.sendSpansOk, 1)
+		atomic.AddInt64(&r.stats.spansSent, int64(len(spans)))
 	} else {
-		r.koSend++
+		atomic.AddInt64(&r.stats.sendSpansKo, 1)
+		atomic.AddInt64(&r.stats.spansNotSent, int64(len(spans)))
 	}
 
 	return lastError, shouldExit
 }
 
+// Stop recorder
+func (r *SpanRecorder) Stop() {
+	if r.debugMode {
+		r.logger.Println("Scope recorder is stopping gracefully...")
+	}
+	r.t.Kill(nil)
+	_ = r.t.Wait()
+	if r.hasSpans() {
+		err, _ := r.SendSpans()
+		if err != nil {
+			r.logger.Printf("error sending spans: %v\n", err)
+		}
+	}
+	if r.debugMode {
+		r.writeStats()
+	}
+}
+
+// Write statistics
+func (r *SpanRecorder) writeStats() {
+	r.statsOnce.Do(func() {
+		r.logger.Printf("** Recorder statistics **\n")
+		r.logger.Printf("  Total spans: %d\n", r.stats.totalSpans)
+		r.logger.Printf("  Test spans: %d\n", r.stats.testSpans)
+		r.logger.Printf("  Spans sent: %d\n", r.stats.spansSent)
+		r.logger.Printf("  Spans not sent: %d\n", r.stats.spansNotSent)
+		r.logger.Printf("  Spans rejected: %d\n", r.stats.spansRejected)
+		r.logger.Printf("  SendSpans calls: %d\n", r.stats.sendSpansCalls)
+		r.logger.Printf("  SendSpans OK: %d\n", r.stats.sendSpansOk)
+		r.logger.Printf("  SendSpans KO: %d\n", r.stats.sendSpansKo)
```

Comment:
Unrelated to this: because we have one agent per module, I think we should write the name of the module we are testing at the beginning of the file, otherwise when testing multiple modules we don't know what we are testing. Maybe dumping the agent metadata should do the trick.

Reply:
Ok, this is the example of the log file when SCOPE_DEBUG is true:

<img width="1015" alt="image" src="https://user-images.githubusercontent.com/69803/74199317-1de1cf80-4c64-11ea-8234-c9bdfecff115.png">

Hopefully this give us enough data to investigate in case of errors.

Reply:
In case of errors, we see more data, like this case:

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/69803/74199740-52a25680-4c65-11ea-98f5-bfa97095482b.png">